azurerm/internal/services/compute/validation.go

```diff
@@ -82,7 +82,7 @@ func ValidateLinuxComputerName(i interface{}, k string, maxLength int) (warnings
 		errors = append(errors, fmt.Errorf("%q cannot begin with an underscore", k))
 	}
 
-	if strings.HasSuffix(v, ".") || strings.HasSuffix(v, "-") {
+	if strings.HasSuffix(v, ".") || (strings.HasSuffix(v, "-") && !allowDashSuffix) {
 		errors = append(errors, fmt.Errorf("%q cannot end with an period or dash", k))
 	}
```

Comment:
to ensure this error message is accurate, could we split this into two checks:

```suggestion
	if strings.HasSuffix(v, ".") {
 		errors = append(errors, fmt.Errorf("%q cannot end with an period", k))
 	}
 	if strings.HasSuffix(v, "-") && !allowDashSuffix {
 		errors = append(errors, fmt.Errorf("%q cannot end with a dash", k))
 	}
```